packages/mdc-segmented-button/segment/test/foundation.test.ts

```diff
@@ -0,0 +1,156 @@
+    mockAdapter.isSingleSelect.and.returnValue(singleSelect);
+
+    // Need calls to mocked methods to change mocked state
+    mockAdapter.addClass.and.callFake((cssClass: string) => {
+      mockAdapter.hasClass.withArgs(cssClass).and.returnValue(true);
+    });
+    mockAdapter.removeClass.and.callFake((cssClass: string) => {
+      mockAdapter.hasClass.withArgs(cssClass).and.returnValue(false);
+    });
+
+    return {foundation, mockAdapter};
+  };
+
+  const setupSelectedTest = (singleSelect: boolean = false) => {
+    return setupTest(true, singleSelect);
+  };
+
+  const setupUnselectedTest = (singleSelect: boolean = false) => {
+    return setupTest(false, singleSelect);
+  };
+
+  it('#isSelected returns true if segment is selected', () => {
+    const {foundation} = setupSelectedTest();
+    expect(foundation.isSelected()).toBeTruthy();
+  });
+
+  it('#isSelected returns false if segment is unselected', () => {
+    const {foundation} = setupUnselectedTest();
+    expect(foundation.isSelected()).toBeFalsy();
+  });
+
+  it('#getSegmentId returns segment id', () => {
+    const {foundation, mockAdapter} = setupSelectedTest();
+    mockAdapter.getAttr.withArgs('data-segment-id').and.returnValue(SEGMENT_ID);
+    expect(foundation.getSegmentId()).toEqual(SEGMENT_ID);
+  });
+
+  describe('Selection toggling', () => {
+    it('#setSelected adds the `selected` css class and if singleSelect then sets aria-checked to true', () => {
+      const {foundation, mockAdapter} = setupUnselectedTest(true);
+      foundation.setSelected();
+
+      expect(mockAdapter.addClass).toHaveBeenCalledWith(SELECTED_CLASS);
+      expect(mockAdapter.setAttr).toHaveBeenCalledWith('aria-checked', 'true');
+    });
+
+    it('#setSelected adds the `selected` css class and if not singleSelect then sets aria-pressed to true', () => {
+      const {foundation, mockAdapter} = setupUnselectedTest();
+      foundation.setSelected();
+
+      expect(mockAdapter.addClass).toHaveBeenCalledWith(SELECTED_CLASS);
+      expect(mockAdapter.setAttr).toHaveBeenCalledWith('aria-pressed', 'true');
+    });
+
+    it('#setUnselected removes the `selected` css class and if singleSelect then sets aria-checked to false', () => {
+      const {foundation, mockAdapter} = setupSelectedTest(true);
+      foundation.setUnselected();
+
+      expect(mockAdapter.removeClass).toHaveBeenCalledWith(SELECTED_CLASS);
+      expect(mockAdapter.setAttr).toHaveBeenCalledWith('aria-checked', 'false');
+    });
+
+    it('#setUnselected removes the `selected` css class and if not singleSelect then sets aria-pressed to false', () => {
+      const {foundation, mockAdapter} = setupSelectedTest();
+      foundation.setUnselected();
+
+      expect(mockAdapter.removeClass).toHaveBeenCalledWith(SELECTED_CLASS);
+      expect(mockAdapter.setAttr).toHaveBeenCalledWith('aria-pressed', 'false');
+    });
+  });
+
+  describe('Click handling', () => {
```

Comment:
It would be useful to group this by feature instead of method names.

For example, `single selection`, `multi selection`, etc.